config-syncer updates

SentinelDev1 opened the pull request:
Created new PR given the old one was inadvertently closed and merged.

SentinelDev1 committed:
Merge pull request #33 from gmx-io/updates

Merged updates branch into config-syncer branch

SentinelDev1 committed:
Merge pull request #34 from gmx-io/updates

Update keys.ts

SentinelDev1 committed:
Merge pull request #35 from gmx-io/updates

add GLV_SHIFT_GAS_LIMIT to Config

SentinelDev1 committed:
Update ConfigSyncer.ts

Minor change to make error message delivery more in line with other tests and general testing norms.

SentinelDev1 committed:
Update riskOracle.ts

Updated to contain RiskOracleConfig data used to enable/disable markets, parameters and market parameter combinations.

SentinelDev1 committed:
Update types.d.ts

Updated for changes to the config/riskOracle.ts file

SentinelDev1 committed:
Update deployConfigSyncer.ts

Updated for changes to the config/riskOracle.ts file

SentinelDev1 committed:
Update updateRiskOracleConfig.ts

Refactored and updated to take the markets, parameters and market parameters being updated from the config/riskOracle.ts file

gdev8317 commented on scripts/updateRiskOracleConfig.ts:
```diff
@@ -1,51 +1,85 @@
   console.info("multicallWriteParams", multicallWriteParams);
 
-  if (process.env.WRITE) {
+  if (multicallWriteParams.length > 0 && process.env.WRITE === "true") {
```
can it be similar to updateMarketConfigUtils?

https://github.com/gmx-io/gmx-synthetics/blob/updates/scripts/updateMarketConfigUtils.ts#L555

there is only one check `if (write) ...`

SentinelDev1 replied:
Sure, the additional checks were mainly helpful for debugging but I will remove them and better align with the provided example

gdev8317 commented on scripts/updateRiskOracleConfig.ts:
```diff
@@ -1,51 +1,85 @@
+
+  const syncConfigKeys = [];
+  
+  for (const marketAddress in riskOracleConfig.markets) {
```
you can use `for (const [marketAddress, marketConfig] of Object.entries(riskOracleConfig.markets) { ... }`

it's better to not use `for in` to avoid unexpected behavior (for example, with arrays it will return both elements and some array properties)

SentinelDev1 replied:
Thanks for the tip, that makes sense, I will update accordingly.

gdev8317 commented on config/riskOracle.ts:
```diff
@@ -1,22 +1,74 @@
+      markets: {
+        "0xD996ff47A1F763E1e55415BC4437c59292D1F415": { // Address for current Avalanche Fuji deployment AVAX/USDC pool
+          syncConfigMarketDisabled: false,
+          marketParameters: {
```
wondering maybe can be just "parameters"? it's already inside "markets", so it should be clear that it belongs to markets?

SentinelDev1 replied:
Ok, sure I can update, I was just afraid it might be confusing given the ability to disable parameters for all markets as well.

gdev8317 commented on scripts/updateRiskOracleConfig.ts:
```diff
@@ -1,51 +1,85 @@
+  
+  for (const marketAddress in riskOracleConfig.markets) {
+    const marketConfig = riskOracleConfig.markets[marketAddress];
+    syncConfigKeys.push({
```
could you check how we handle these in e.g. updateMarketConfig https://github.com/gmx-io/gmx-synthetics/blob/updates/scripts/updateMarketConfigUtils.ts#L507

it first reads all values from DataStore https://github.com/gmx-io/gmx-synthetics/blob/updates/scripts/updateMarketConfigUtils.ts#L528

then it updates params if only it doesn't match values in DataStore https://github.com/gmx-io/gmx-synthetics/blob/updates/scripts/updateMarketConfigUtils.ts#L548

without this current script will always send transaction

SentinelDev1 replied:
Ok, but I already included a check for this in updateRiskOracleConfig.ts: 
https://github.com/gmx-io/gmx-synthetics/blob/config-syncer/scripts/updateRiskOracleConfig.ts#L52-L65 

This is similar to how it was done in updateOracleConfig.ts: 
https://github.com/gmx-io/gmx-synthetics/blob/config-syncer/scripts/updateOracleConfig.ts#L19-L23

Maybe I'm missing something, but isn't this already addressed with the highlighted code? Or maybe the issue is that I'm doing this check after retrieving all the markets, market parameters and parameters from riskOracle.ts?

SentinelDev1 committed:
Moved the logic previously in updateRiskOracleConfig.ts to updateRiskOracleConfigUtils.ts to better align with how other scripts are structured.

Updated riskOracle.ts and updateRiskOracleConfigUtils.ts per the feedback from G in the PR and now most if not all comments should be addressed.

SentinelDev1 committed:
Updated to compute oldValue in scripts/updateRiskOracleConfigUtils.ts using a clearer method

SentinelDev1 committed:
Merge branch 'updates' into config-syncer
merging latest changes to updates branch into the config-syncer branch

SentinelDev1 committed:
Updated test/config/ConfigSyncer.ts and utils/configSyncer.ts

-Added 1 test from the Guardian audit's test script to the ConfigSyncer.ts script with some minor adjustments and renamed for consistency with other 'revert' test names.
-Added additional 'expect' checks from a Guardian test to ConfigSyncer.ts tests where ConfigSyncer.sync() doesn't revert
-Minor clean-up and reorganization of ConfigSyncer.ts and configSyncer.ts
